src/expr/src/relation/mod.rs

```diff
@@ -1194,12 +1194,56 @@ impl MirRelationExpr {
         aggregates: Vec<AggregateExpr>,
         expected_group_size: Option<usize>,
     ) -> Self {
-        MirRelationExpr::Reduce {
-            input: Box::new(self),
-            group_key: group_key.into_iter().map(MirScalarExpr::Column).collect(),
-            aggregates,
-            monotonic: false,
-            expected_group_size,
+        // Check if we have an unsigned aggregation and add an appropriate cast here
+        // when the output is expected to be unsigned. This is done since rendering
+        // needs to accumulate to a signed number to deal with retractions, and we do
+        // not want to panic there (#17510).
+        let unsigned_sums = aggregates
+            .iter()
+            .enumerate()
+            .filter(|(_, agg)| {
+                agg.func == AggregateFunc::SumUInt16 || agg.func == AggregateFunc::SumUInt32
+            })
+            .map(|(idx, _)| idx)
+            .collect::<Vec<_>>();
```

Comment:
I think we can remove this `collect` and replace it by [`peekable`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.peekable).

Reply:
Sounds good, thanks!